### PR TITLE
Menu item API

### PR DESCRIPTION
**Pull request by tehsmeely**
Exposes an api in `System` for adding (etc) menu items.

Slightly painful since it involves passing callbacks into C but it's handled by double-boxing dynamic Fn() trait objects.

There's some degree of code around handling the possible kinds of MenuItem values (as MenuItemKind).
Potentially one could go a step further so users do not have the directly deal with the index value and can get
its typed (bool or string) accordingly.

Adds a new example that adds one of each menu item

### CODE REVIEW

File: src/system.rs [hunk 1]
```diff
@@ -1,14 +1,25 @@
...
     cstr_core::CString,
 };
 
-use crankstart_sys::ctypes::c_int;
-pub use crankstart_sys::PDButtons;
-use crankstart_sys::{PDDateTime, PDLanguage, PDPeripherals};
-
 static mut SYSTEM: System = System(ptr::null_mut());
 
+static mut MENU_ITEM_CALLBACKS: Vec<Box<dyn Fn()>> = Vec::new();
```
Comment (boozook) [inline src/system.rs:19]: Not sure I understand how it used. Could you explain please?

**Comment by boozook** [comment]
@tehsmeely, thank you for your attempt to make the better implementation! 👍

Just one question. Have you seen [another attempt][my]? Could you please compare yours with that?

One of the advantages that I see here is that there are no generics, which makes it easier to use. However, the UserData type is lost here, which is a disadvantage.

[my]: https://github.com/boozook/playdate/blob/main/api/menu/src/lib.rs#L267-L282

File: src/system.rs (hunk 1 above)
Comment (tehsmeely) [reply src/system.rs]: Oops sorry, poor review on my part these are left over from original implementations. I've removed them now in this PR

**Comment by tehsmeely** [comment]
> @tehsmeely, thank you for your attempt to make the better implementation! 👍
> 
> Just one question. Have you seen [another attempt](https://github.com/boozook/playdate/blob/main/api/menu/src/lib.rs#L267-L282)? Could you please compare yours with that?
> 
> One of the advantages that I see here is that there are no generics, which makes it easier to use. However, the UserData type is lost here, which is a disadvantage.

As far as I'm aware there's little to no need for userdata handling here as that's primarily the C handling for generic callback functions. Since we're using rust closures in this implementation, we can include any relevant userdata into that thunk and continue from there.

**Comment by tehsmeely** [comment]
I just pushed a fix for a soundness issue where one could drop a MenuItem and result in the item still being visible in the menu, at which point pressing would result in reading of unmanaged memory. 

Worked around this by forcing the "removeMenuItem" call in the drop impl for MenuItem instead. The `remove_menu_item` method still exists but it's just an explicit drop. 

`removeAllMenuItems` is gone because it doesn't make sense as it would leave MenuItems with dangling pointers. This is not a problem from the API since it's easier to just drop MenuItems

**Review by boozook — APPROVED** [review]

**Pull request merged**